modules/client/src/main/java/org/apache/ignite/internal/client/ReliableChannel.java

```diff
@@ -754,8 +755,8 @@ class ClientChannelHolder {
         /** Address that holder is bind to (chCfg.addr) is not in use now. So close the holder. */
         private volatile boolean close;
 
-        /** Timestamps of reconnect retries. */
-        private final long[] reconnectRetries;
+        /** Reconnect retry counter. */
+        private final AtomicLong reconnectRetries = new AtomicLong();
```

Comment:
I don't think we use it anywhere anymore?

Reply:
Good catch, I've started refactoring the throttling logic initially, then realized we don't need it.

Fixed.